lib/components/form/call-taker/date-time-picker.tsx

```diff
@@ -0,0 +1,223 @@
+
+  const intl = useIntl()
+
+  /**
+   * Parse a time input expressed in the agency time zone.
+   * @returns A date if the parsing succeeded, or null.
+   */
+  const parseInputAsTime = (
+    timeInput: string = getCurrentTime(homeTimezone),
+    date: string = getCurrentDate(homeTimezone)
+  ) => {
+    if (!timeInput) timeInput = getCurrentTime(homeTimezone)
+
+    // Match one of the supported time formats
+    const matchedTimeFormat = SUPPORTED_TIME_FORMATS.find((timeFormat) =>
+      isMatch(timeInput, timeFormat)
+    )
+    if (matchedTimeFormat) {
+      const resolvedDateTime = format(
+        parse(timeInput, matchedTimeFormat, new Date()),
+        'HH:mm:ss'
+      )
+      return toDate(`${date}T${resolvedDateTime}`)
+    }
+    return ''
+  }
+
+  const dateTime = parseInputAsTime(time, date)
+
+  // Handler for setting the query parameters
+  useEffect(() => {
+    if (safeFormat(dateTime, OTP_API_DATE_FORMAT, {}) !== '') {
+      setQueryParam({
+        date: safeFormat(dateTime, OTP_API_DATE_FORMAT, {
+          timeZone: homeTimezone
+        }),
+        departArrive,
+        time: safeFormat(dateTime, OTP_API_TIME_FORMAT, {
+          timeZone: homeTimezone
+        })
+      })
+    }
+  }, [dateTime, departArrive, homeTimezone, setQueryParam])
+
+  // Handler for updating the time and date fields when NOW is selected
+  useEffect(() => {
+    if (departArrive === 'NOW') {
+      setTime(getCurrentTime(homeTimezone))
+      setDate(getCurrentDate(homeTimezone))
+    }
+  }, [departArrive, setTime, setDate, homeTimezone])
+
+  const unsetNow = () => {
+    if (departArrive === 'NOW') setDepartArrive('DEPART')
+  }
+
+  return (
+    <>
+      <select
+        onBlur={(e) => setDepartArrive(e.target.value)}
+        onChange={(e) => setDepartArrive(e.target.value)}
+        onKeyDown={onKeyDown}
+        value={departArrive}
+      >
+        {getDepartureOptions(intl).map(({ text, value }) => (
+          <option key={value} value={value}>
+            {text}
+          </option>
+        ))}
+      </select>
+      <OverlayTrigger
+        overlay={
+          <Tooltip id="time-tooltip">
+            {safeFormat(dateTime, timeFormat, {
+              timeZone: homeTimezone
+            })}
+          </Tooltip>
+        }
+        placement="bottom"
+        trigger={['focus', 'hover']}
+      >
+        <input
+          className="datetime-slim"
+          onChange={(e) => {
+            setTime(e.target.value)
+            unsetNow()
+          }}
+          onFocus={(e) => e.target.select()}
+          onKeyDown={onKeyDown}
+          style={{
+            fontSize: 'inherit',
+            lineHeight: '.8em',
+            marginLeft: '3px',
+            padding: '0px',
+            width: '50px'
+          }}
+          // Don't use intl.formatTime, so that users can enter time in 12hr or 24hr format at their leisure.
+          value={time || format(dateTime, 'H:mm', { timeZone: homeTimezone })}
```

Comment:
This is causing a tiny bug where the input refuses to be empty
![timeloop](https://user-images.githubusercontent.com/115499534/207437368-f0e7b8b7-dd2b-4e2e-a1b9-0d7381653eb1.gif)